config/Config-devel.in

```diff
@@ -17,6 +17,30 @@ menuconfig DEVEL
 		  Store built firmware images and filesystem images in this directory.
 		  If not set, uses './bin/$(BOARD)'
 
+	choice
+		prompt "Preferred Download tool" if DEVEL
+		default DOWNLOAD_TOOL_PREFER_ARIA2C
+		help
+		  Set the preferred tool to use to download package on the buildroot.
+		  If the tool is not present in the system, the system will fallback
+		  to the others until it find one.
```

Comment:
Do we really need to clutter the build system with such logic? That list might get miles long soon, so what about simply admitting, that we support sane defaults, `wget` and `curl` out of the box and treat any other tool as a custom downloader? So if user wants to use tool `foo`, then simply just use that. If I explicitly enable tool `foo`, then simply error out if it's not available, no need for fallback handling, keep it simple.

Reply:
I agree that curl should return as the default, but some people need aria

I don't think that there even exists that many tools for just doing a download so the list would not get much larger. However, I didn't even know that aria existed so...

Reply:
I missed the conversation, why do some people need aria?

Reply:
Mainly Chinese users that have slow download connection and mirrors problem. Aria2c can check every mirror and select a good one and increase download speed.